code/game/objects/items/devices/multitool.dm

```diff
@@ -21,6 +21,19 @@
 	var/hack_speed = 10 SECONDS // Only used for vendors right now
 	var/next_scan
 
+	var/list/encryption_keys = list()
+
+/obj/item/device/multitool/proc/load_encryption_key(var/key, var/obj/object)
+	encryption_keys[key] = object
```

Comment:
possibly make this a weakref instead? This seems like it could very easily make harddels happen